Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -0,0 +1,1032 @@
+            return false;
+        TString hname = TString::Format("%s",name.c_str());
+        TString htit = TString::Format("%s",title.c_str());
+//BASELINE
+        if ( type.compare("BASELINE")== 0){
+    // Check if Histogram Exist
+            if (std::find(fNameHisto.begin(), fNameHisto.end(), name) == fNameHisto.end()) {
+                fNameHisto.push_back(name);
+                ih = fMapBinNameIdSummaryHisto.size();
+                fMapBinNameIdSummaryHisto.insert(std::pair<std::string,int>(chName,ih));
+                h1d.histo=new TH1F(hname, htit, fNumBinX, fMinBinX,fMaxBinX);
+                h1d.condHisto.push_back(condition);
+                ih = (int)fMatrixHistoBaseline[mod][ch].size();
+                fMatrixHistoBaseline[mod][ch].push_back(h1d);
+
+                if (ih < (int)fMatrixHistoBaseline[mod][ch].size()){
+                    getObjectsManager()->startPublishing(fMatrixHistoBaseline[mod][ch].at(ih).histo);
+                    try {
+                        getObjectsManager()->addMetadata(fMatrixHistoBaseline[mod][ch].at(ih).histo->GetName(), "custom", "34");
+                        return true;
+                    }
+                    catch (...) {
+                        ILOG(Warning, Support) << "Metadata could not be added to " << fMatrixHistoBaseline[mod][ch].at(ih).histo->GetName() << ENDM;
+                        return false;
+                    }
+
+                    delete h1d.histo;
+                    h1d.condHisto.clear();
+                }
+                return true;
+            }
+            else{
+                for (int i=0; i< (int)fMatrixHistoBaseline[mod][ch].size(); i++){
+                    fMatrixHistoBaseline[mod][ch].at(i).histo->Reset();
+                    }
+               return true;
+            }
+        }
+//COUNTS
+        if ( type.compare("COUNTS")== 0){
+    // Check if Histogram Exist
+            if (std::find(fNameHisto.begin(), fNameHisto.end(), name) == fNameHisto.end()) {
+                fNameHisto.push_back(name);
+                h1d.histo=new TH1F(hname, htit, fNumBinX, fMinBinX,fMaxBinX);
+                h1d.condHisto.push_back(condition);
+                ih = (int)fMatrixHistoCounts[mod][ch].size();
+                fMatrixHistoCounts[mod][ch].push_back(h1d);
+
+                if (ih < (int)fMatrixHistoCounts[mod][ch].size()){
+                    getObjectsManager()->startPublishing(fMatrixHistoCounts[mod][ch].at(ih).histo);
+                    try {
+                        getObjectsManager()->addMetadata(fMatrixHistoCounts[mod][ch].at(ih).histo->GetName(), "custom", "34");
+                        return true;
+                    }
+                    catch (...) {
+                        ILOG(Warning, Support) << "Metadata could not be added to " << fMatrixHistoCounts[mod][ch].at(ih).histo->GetName() << ENDM;
+                        return false;
+                    }
```

Comment:
```suggestion
```
you probably don't need this, it was an example

Reply:
Ok i'll remove it.